### ISSUE TITLE
bug: patched YT reboots system every day on the same time

### ISSUE THREAD

**Issue by Atexor** [revanced/revanced-patches#1543]
### Type

Error at runtime

### Bug description

Hello,
I am using Android 13 on custom AOSP (POCO F4), all permissions granted. YT version to patch: 17.49.37.

Every time around 22:01-22:02 my phone is rebooting. 

Logs shows that it's caused by YT ReVanced, which is installed with root. After Reboot YouTube (ReVanced) app is removed completely, in logs there is also information about it:
`01-23 22:03:40.834 22338 22754 E LoaderCursor: Invalid package removed: com.google.android.youtube [ 353 ]`

I have also prepared by ReVanced Manager patched .apk file with the same patched (described under "steps to reproduce"). Before install apk I am getting question whether to update app over installed "null". Installation ends with error due to conflict despite I cannot find and YT remains. Play Store/[De-Bloater](https://github.com/sunilpaulmathew/De-Bloater) don't detect YT but there are remains because after installing vanilla YT 17.49.37, both ReVanced and YT settings are restored.

I thought it might be related to with turning off Ambient of Display at 22:00 (this time my phone wokes and then blanks screens after a while), but after crash on 22:00 I tried to repeat it by changing disabling AoD time to 22:30, installing YT Revanced again but crash didn't appear again.

Except that YT Revanced works pretty fine.

### Steps to reproduce

1. Install YouTube 17.49.37.
2. Patch:
> Custom branding
> Custom video buffer
> Custom video speed
> Disable startup shorts player
> Downloads
> General ads
> Minimized playback
> Premium heading
> Return youtube dislike
> Theme
> Video ads
3. Install as root.
4. Wait to 22:00-22:03.
5. System reboots with removing YT. Logcat logs starts from fatal error, not before.

### Relevant log output

```shell
https://pastebin.com/hfgJ5jpb
```

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by oceangith** [comment]
Same problem with me; YouTube app and YouTube Music

**Comment by Atexor** [comment]
Yesterday I tried to change AoD turning off to 23:00 and removed almost all patch except one - no crash. Today I used YT Revanced with 10 patches (except Custom Video Speed) and got the same issue. Few minutes after 23:00, so that issue is definitely related with AoD and one of these 10 patches.

New logs (looks a bit different than previous):
https://pastebin.com/XaFZ5xnq

I might try to test it by removing next patches next dayts but it will take time. I tried to change AoD time to closer time but I was unable to replicate the problem. I'm not sure if the phone doesn't need a much longer deep sleep state.

**Comment by oSumAtrIX** [comment]
Can't you just set the time on your phone?

**Comment by xgangsa** [comment]
Mine random reboot and uninstall automatically after install revanced root version, this happen in end of 2022 when i install revanced or vanced version. 

This because package name in app match with youtube official, playstore detect from that and force uninstall via random reboot.

But now i install revanced with custom package name without root version and microG. no random reboot, no random force uninstall and i happy with it.

**Comment by oSumAtrIX** [comment]
@xgangsa PlayStore does not detect or remove mounts.

**Comment by xgangsa** [comment]
> @xgangsa PlayStore does not detect or remove mounts.

So from where playstore can uninstalled vanced or revanced after random reboot? I already detach it from play store with magisk module. This is the same when i install revanced or vanced version without root or non root version.

I think this problem come from app package name. Now i never have random reboot again. Revanced still there whenever i reboot my device manually.

Sorry, my english skill so bad.

**Comment by oSumAtrIX** [comment]
The mount script is likely failing.

**Comment by ILoveOpenSourceApplications** [comment]
Is this resolved @oSumAtrIX ?

**Comment by oSumAtrIX** [comment]
I don't have this issue, can't say.

**Comment by ILoveOpenSourceApplications** [comment]
@Atexor or @oceangith are you guys still having this issue?

**Comment by Atexor** [comment]
@oSumAtrIX @ILoveOpenSourceApplications I've recently clean flashed new system without pre-installed YT. I installed it (recomenned v18.03.6 from apkmirror) and patched. That issue seems to be gone (reboots on the same time), however I noticed that (probably) yesterday YT was deleted and when I installed it again I got question about updating "null" app as described on main post.

I can't find in logcat nothing related to YT in recent time.

Also, I noticed that quite often when I am using "custom brading" patch, icon of YouTube remains old, even after reboot/launcher restart (I use Nova Launcher). Sometimes it changes after longer time, sometimes immediately after patching. I will try to take a look on it in next days and catch logs if possible.

**Comment by Atexor** [comment]
No, device has normal dark theme with "monet" accents (changing some colors of buttons depending of colors on wallpaper).

I mean this kind of icon change (left from internet, right is currently mine):
![obraz](https://user-images.githubusercontent.com/34380553/221540915-bf9a2e7f-714c-4fd8-95fd-789b525322b6.png)

Sometimes icon changes to that on right, sometimes remains stock. So I wonder whether it's cause of branding patch which is not working as expected and system wants restore original YT.

Maybe it would be good to add in that patch chaning package name. It should both detach YouTube from Playstore (prevent from auto-updated/manual updates by mistake) and maybe prevent system from restoring/deleting patched YT.
